src/repos/ArticleRepo.ts

```diff
@@ -83,6 +84,81 @@ const getArticlesByPublicationSlugs = async (
     });
 };
 
+const getShuffledArticlesByPublicationSlugs = async (
+  slugs: string[],
+  limit: number = DEFAULT_LIMIT,
+  offset: number = DEFAULT_OFFSET,
+  timeRange = 12, // int value in months before current date to get articles from
+): Promise<Article[]> => {
+  // return a list of ArticleModels sorted by date, but each publisher has to show up at least once
+  // before articles by the same publisher shows up again
+  // e.g. publisher 1 has articles 1.1 (published jan 1), 1.2 (published jan 6)
+  //      publisher 2 has articles 2.1 (published jan 5), 2.2 (published jan 4)
+  //      publisher 3 has articles 3.1 (published jan 2), 3.2 (published jan 3)
+  // the function should return [1.1, 3.1, 2.1, 3.2, 2.2, 1.2]
```

Comment:
This should ideally be a docstring, anybody calling this function should be able to see this information, so it's better to structure this as a docstring with an `@returns` than as multiple single-line comments